AutoHub.BLL/Services/UserService.cs

```diff
@@ -1,63 +1,95 @@
 using System;
 using System.Collections.Generic;
-using System.Security.Cryptography;
-using System.Text;
-using System.Text.RegularExpressions;
+using System.Linq;
+using AutoHub.BLL.DTOs.UserDTOs;
 using AutoHub.BLL.Interfaces;
 using AutoHub.DAL.Entities;
+using AutoHub.DAL.Enums;
 using AutoHub.DAL.Interfaces;
+using AutoMapper;
 
 namespace AutoHub.BLL.Services
 {
     public class UserService : IUserService
     {
+        private readonly IAuthService _authService;
+        private readonly IMapper _mapper;
         private readonly IUnitOfWork _unitOfWork;
 
-        public UserService(IUnitOfWork unitOfWork)
+        public UserService(IUnitOfWork unitOfWork, IMapper mapper, IAuthService authService)
         {
             _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _authService = authService;
         }
 
-        public IEnumerable<User> GetAll()
+        public IEnumerable<UserResponseDTO> GetAll()
         {
-            return _unitOfWork.Users.GetAll();
+            var users = _unitOfWork.Users.GetAll();
+            var mappedUsers = _mapper.Map<IEnumerable<UserResponseDTO>>(users);
+            return mappedUsers;
         }
 
-        public User GetById(int id)
+        public UserResponseDTO GetById(int userId)
         {
-            return _unitOfWork.Users.GetById(id);
+            var user = _unitOfWork.Users.GetById(userId);
+            var mappedUser = _mapper.Map<UserResponseDTO>(user);
+            return mappedUser;
         }
 
-        public bool Register(User userModel)
+        public UserResponseDTO GetByEmail(string email)
         {
-            if (!IsPasswordMatchRules(userModel.Password) || !IsEmailUnique(userModel.Email))
-                return false;
-            _unitOfWork.Users.Add(userModel);
-            _unitOfWork.Commit();
-            return true;
+            var user = _unitOfWork.Users.Find(user => user.Email == email).FirstOrDefault();
+            var mappedUser = _mapper.Map<UserResponseDTO>(user);
+            return mappedUser;
         }
 
-        public bool Login()
+        public UserLoginResponseDTO Login(UserLoginRequestDTO userModel)
         {
-            throw new NotImplementedException();
+            var user = _unitOfWork.Users.Find(user1 =>
+                user1.Email == userModel.Email).FirstOrDefault();
+
+            if (user == null || !_authService.VerifyPassword(userModel.Password, user.Password)) return null;
```

Comment:
this condition should throw an exception instead of returning null
also it would be better to extract _authService.VerifyPassword(userModel.Password, user.Password) to a separate variable